### ISSUE TITLE
Add option for tabs to fill the tab bar

### ISSUE THREAD

**Issue by Jacalz** [fyne-io/fyne#1121]
### Is your feature request related to a problem? Please describe:
I think that it might be a good idea to have a setting for the tab bar to let the tabs fill the space instead of being small. I am talking about something in line of the tabs in for example Nautilus (Gnome file manager) or Finder (on macOS) where tabs generally fill the space that they have to work with:

![image](https://user-images.githubusercontent.com/25466657/85222723-dc092880-b3bd-11ea-8c49-f25c4c6825e4.png)

![image](https://user-images.githubusercontent.com/25466657/85222743-fc38e780-b3bd-11ea-8fb8-68161058feec.png)

### Is it possible to construct a solution with the existing API?
In general yes it probably should with some changes to the layout code. It might however be necessary to have #371 to make it better with more tabs.

### Describe the solution you'd like to see:

Possibly having a field `tab.FillSpace = true` to enable it or something along those lines.

**Comment by andydotxyz** [comment]
This seems like a style choice rather than functionality or a use-case?
Maybe the current setup is not ideal and can be improved, but not sure if it needs to be an API for developer to choose?

**Comment by Jacalz** [comment]
Yeah I think you are right about that. I will go ahead and close this.

**Comment by stuartmscott** [comment]
Reopening as I find this "stretch to fill" style desirable, tho I agree it should not be an API for devs.

Probably depends on #1310

**Comment by itsmewes** [comment]
I was looking through the GitHub issues for just this. Is there any update on this by any chance? Sorry not trying to push or anything, this is the only place I've really found exactly what I'm looking for and not sure if this has been implemented yet or if there is a way to currently do this in v2.

Really enjoying getting into Fyne at the moment.

**Comment by andydotxyz** [comment]
Now that `Tabs` are split to `AppTabs` and `DocTabs` I think we could clarify that this (or does it?) applies to `DocTabs`.
The reason I say that is that `AppTabs` are typically adjusted per-platform but doctabs are a more abstract concept.

**Comment by atljoseph** [comment]
Stretch to fill should be available parameter.

**Comment by atljoseph** [comment]
How about just add func SetStretchToFit(val bool) to both AppTabs and DocTabs structs? It would just use the mobile layout for the AppTabs/DocTabs widgets. 

But then there is a question of theme usage. Seems that the mobile layout forces a thicker tab bar and bigger icons.

**Comment by andydotxyz** [comment]
> Seems that the mobile layout forces a thicker tab bar and bigger icons. So maybe a SetTabBarThickness(val foat32), and scale the content of each tab proportional to the thickness.

That seems to be tangential to the request at the top of this issue. Asking things to fill is one request, dynamically changing the line height for one of the containers can have other ramifications and should be considered a separate issue I would think.

**Comment by atljoseph** [comment]
Depends on the depth at which one considers side-effects “separate” or
related. I certainly would not call it tangential by any means (I have a
physics degree, actually). If by stretching to fit, the bar becomes thicker
and recognizes different rules of theming (and much it does), then somewhat
“slightly integral but not mission critical” would be more like it. It all
depends on how you plan to solve the issue. It could be tangential,
depending on how you see the solution. All I’ve seen so far is “it has to
be this way” or “that won’t work”. I’m not sure I have anything to
contribute here that is perceived as valuable, so I’m gonna buzz out.

**Comment by andydotxyz** [comment]
By no means did I mean that a side effect was separate to this. What I meant is that the request seems to be for tabs to fill the bar instead of be minimum width - so adjusting the height should be completely separate.

If the implementation means the height changes that of course is not tangential - it is completely unexpected. I think this demonstrates the pitfalls of "it's just a line of code" when trying to expose internal implementation details as API features.

**Comment by atljoseph** [comment]
And the pitfalls of purism.

**Comment by atljoseph** [comment]
I’m still caught up in the fact it’s been requested for 3 years…. Because we can’t figure out how to “fit it in”. Nothing in life is perfect. Not people. Not libraries.

**Comment by andydotxyz** [comment]
You are welcome to help us find the right solution - but the rude words you have used to describe the current state of affairs is off-putting.
Perhaps perfection is not worth chasing - but a consistent API that is discoverable and meets our design guidelines is critical to keep this project manageable.

**Comment by atljoseph** [comment]
There is rude and then there is raw/real/true/human. Here, the juice isn’t
worth the squeeze. And that’s ok, whether it’s an opinion or a fact. Life
has more to offer than code. Thank you for the lib. Thank you for reminding
me about the important parts of life. Truly.

**Comment by dweymouth** [comment]
If it's considered undesirable to add an API to the tabs container directly, I propose that the "FeatureTheme" concept introduced in https://github.com/fyne-io/fyne/pull/5239 could be used to allow themes to set whether tabs should expand to fill the space or not

**Comment by andydotxyz** [comment]
It's been a long running ticket so it's hard to recall all the reasons - but if this was to apply only to `DocTabs` maybe it can be an API on the container.
Or it could be a setting - either at toolkit or theme level. I don't have particularly strong feelings, but this seems like user preference.

In an alternative thought, given what I have said above, perhaps they should always just fill the space until there are too many to fit. Will there be anyone upset that the tabs are no longer min-width?

**Comment by dweymouth** [comment]
I don't think we have any reason to just make the change without having a way to choose between the non-fill and fill behavior. Neither option is better than the other, it's all down to user and/or developer preference. So I'd say to avoid code and app behavior changes for 'no reason', we should keep the tabs behavior as-is until introducing some means of choosing whether they expand to fill or not.
